### ISSUE TITLE
New additionalProperties default breaks `oneOf`

### ISSUE THREAD

**Issue by andrenarchy** [drwpow/openapi-typescript#607]
I experience exactly the problem that @radist2s described in https://github.com/drwpow/openapi-typescript/pull/585#issuecomment-840149431:

```json
{
  "pictures": {
    "type": "array",
    "items": {
      "oneOf": [
        {
          "type": "integer"
        },
        {
          "$ref": "#/components/schemas/ItemsArtworksPictures"
        }
      ]
    }
  }
}
```
results in 
```typescript
{
  pictures?: ((number | components["schemas"]["ItemsArtworksPictures"]) & {
    [key: string]: any;
  })[];
}
```
which is useless (and wrong I hope?).

I'm reverting to `3.2.3` for now but it'd be great to have this fixed soon! :)

**Comment by drwpow** [comment]
Going to fix this in the next release. Thanks for raising!

**Comment by drwpow** [comment]
Released in `3.3.0`!

**Comment by andrenarchy** [comment]
Thank you, works as expected! :)
